### PR TITLE
adding deployment facts

### PR DESCRIPTION
**Pull request by zikalino**
##### SUMMARY
Adding deployment facts module.

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
azure_rm_deployment_facts

### CODE REVIEW

File: lib/ansible/modules/cloud/azure/azure_rm_deployment_facts.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+EXAMPLES = '''
+  - name: Get instance of Deployment
+    azure_rm_deployment_facts:
+      resource_group: myResourceGroup
+      name: myDeployment
+'''
+
+RETURN = '''
+deployments:
+    description: A list of dictionaries containing facts for Artifact.
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_deployment_facts.py]: not aligned description

File: lib/ansible/modules/cloud/azure/azure_rm_deployment_facts.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+            type: complex
+            contains:
+                id:
+                    description:
+                        - Resource id.
+                    returned: always
+                    type: str
+                    sample: "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxxx/resourceGroups/myResourceGroup/providers/Microsoft.Network/networkI
+                             nterfaces/myNetworkInterface"
+                name:
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_deployment_facts.py:104]: is the name and type redundant? if not, then resource group should also been added.

Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_deployment_facts.py:104]: it's just coming from the api, so I have added it. maybe it's not necessary at all.
anyway, the resource group will be the same as deployment resource group.
I believe no ARM deployment creates resources outside it's resource group.

**Review by yungezz — APPROVED** [review]

**Comment by yungezz** [comment]
shipit

**Pull request merged**